plone/app/layout/navigation/navtree.py

```diff
@@ -147,7 +146,7 @@ def buildFolderTree(context, obj=None, query={},
         query['path'] = rootPath
     elif rootPath is None:
         pathQuery = query['path']
-        if type(pathQuery) == StringType:
+        if isinstance(pathQuery, six.text_types):
```

Comment:
This gives an error. `six` has `text_type`, `binary_type` or `string_types`.
With `binary_type` it would be the same as it currently is, but it seems to me that `string_types` would be okay here.

Reply:
We have a graph of Python objects. So the names of those objects, and hence their path, is a native string (binary in Python 2, text in Python 3). In Plone code we need to decide whether we enforce input of the correct type, or encode/decode implicitly. The latter feels too much like what Python 2 does which leads to unexpected encode/decode errors. But since we use utf-8 everywhere it's probably not a big deal, and may be helpful for backwards compatibility with code that was never careful about text vs bytes.

I'm not really advocating for any particular way of handling this, just offering food for thought.

Reply:
I went with six.string_types here but I guess we should put down some guidelines for Plone regarding the use of bytes and text in the comment that seem to become the docs for that (https://github.com/plone/Products.CMFPlone/issues/2184#issuecomment-359445243).